adserver/tasks.py

```diff
@@ -818,6 +865,61 @@ def notify_of_completed_flights():
                 flight
             )
 
+        if flight.auto_renew and not flight.live:
+            # This flight is completed but should be renewed
+            # CONSIDER: Breaking this out into a model method
+            #  although it's hard to re-use this with the
+            #  renewal form since more data is provided there.
+            log.info("Auto-renewing flight %s.", flight)
+
+            new_flight_name = f"{flight.campaign.advertiser} - {timezone.now():%b %Y}"
+            new_flight_slug = slugify(new_flight_name)
+
+            while Flight.objects.filter(slug=new_flight_slug).exists():
+                random_char = get_random_string(1)
+                new_flight_slug = slugify(f"{new_flight_slug}-{random_char}")
+
+            new_flight = Flight(
+                name=new_flight_name,
+                slug=new_flight_slug,
+                start_date=timezone.now().today(),
+                end_date=timezone.now().today() + (flight.end_date - flight.start_date),
```

Comment:
Shouldn't this get the next month or 3 months, and subtract one day to get the end of the month? Seems like we mostly just want these to line up with month boundaries, not a specific number of days. But maybe that's complicated also? 🤷

Reply:
Matching the previous flight length will always work while attempting to align with month boundaries will have weird results if the flight ends mid-month or isn't a month/quarter in length.

Reply:
Sure, but this will consistently lead to weird results of `February 2025` campaign running 2 days into March or similar. Seems like we mostly want it to align with date boundaries, and we just shouldn't run campaign for shorter periods? I feel like this is definitely much less what a user would expect.

Reply:
I don't think we can align with date boundaries easily or consistently. Should a campaign that starts on the 8th run for only 20 days if the next month is February? I acknowledge this is a bit strange but as written, the code functions pretty similarly to the defaults when a staff user hits the "renew" button.

Reply:
After some discussion, let's try launching with this and we'll update it if needed.